src/ui/component/filePicker/filePicker.tsx

```diff
@@ -0,0 +1,186 @@
+import { useCallback, useMemo, useState } from 'react'
+import type { FC } from 'react'
+import { useTranslation } from 'react-i18next'
+import { DragAndDrop } from '@/ui/component/dragAndDrop/dragAndDrop'
+import type { DragDropState } from '@/ui/component/dragAndDrop/dragAndDrop'
+import { DropDownButton } from '@/ui/component/dropDownButton/dropDownButton'
+import type { Option } from '@/ui/component/dropDownButton/dropDownButton'
+import { Icon } from '@/ui/component/icon/icon'
+import { isError } from '@/util/util'
+import classNames from 'classnames'
+import './i18n/index'
+import './filePicker.scss'
+
+export type File = {
+  name: string
+  fullPath: string
+  size: number
+  mediaType: string
+  stream: ReadableStream<Uint8Array>
+}
+
+type FilePickerProps = {
+  onFileChange: (files: File[]) => void
+  hasFolderExplorer?: boolean
+  hasFileExplorer?: boolean
+  multiple?: boolean
+  onError?: (error: Error) => void
+}
+
+// eslint-disable-next-line max-lines-per-function
+export const FilePicker: FC<FilePickerProps> = ({
+  onFileChange,
+  hasFolderExplorer = false,
+  hasFileExplorer = false,
+  multiple = false,
+  onError
+}): JSX.Element => {
+  const [isDraggingOver, setIsDraggingOver] = useState<boolean>(false)
+  const { t } = useTranslation('filePicker')
+
+  const handleDragAndDropState = useCallback((state: DragDropState) => {
+    state === 'dragging-over' ? setIsDraggingOver(true) : setIsDraggingOver(false)
+  }, [])
+
+  const toFileDTO = useCallback(
+    (file: globalThis.File) => ({
+      name: file.name,
+      fullPath: file.webkitRelativePath,
+      size: file.size,
+      mediaType: file.type,
+      stream: file.stream()
+    }),
+    []
+  )
+
+  const handleDrop = useCallback(
+    (event: React.DragEvent<HTMLDivElement>) => {
+      const files = Array.from(event.dataTransfer.items).map(item => {
```

Comment:
I think you can follow same logic by passing a function as an extra arg of `Array.from` as you made in `handleFileChange`.

Reply:
Good remarks 👍 . I've refactored the filepicker and added the closing behaviour in the dropdownbutton component.